runtime/lua/vim/lsp/health.lua

```diff
@@ -28,42 +28,45 @@ local function check_log()
   report_fn(string.format('Log size: %d KB', log_size / 1000))
 end
 
+--- @param f function
+--- @return string
+local function func_tostring(f)
+  local info = debug.getinfo(f, 'S')
+  return ('<function %s:%s>'):format(info.source, info.linedefined)
+end
+
 local function check_active_clients()
   vim.health.start('vim.lsp: Active Clients')
   local clients = vim.lsp.get_clients()
   if next(clients) then
     for _, client in pairs(clients) do
       local cmd ---@type string
-      if type(client.config.cmd) == 'table' then
-        cmd = table.concat(client.config.cmd --[[@as table]], ' ')
-      elseif type(client.config.cmd) == 'function' then
-        cmd = tostring(client.config.cmd)
+      local ccmd = client.config.cmd
+      if type(ccmd) == 'table' then
+        cmd = vim.inspect(ccmd)
+      elseif type(ccmd) == 'function' then
+        cmd = func_tostring(ccmd)
       end
       local dirs_info ---@type string
       if client.workspace_folders and #client.workspace_folders > 1 then
-        dirs_info = string.format(
-          '  Workspace folders:\n    %s',
-          vim
-            .iter(client.workspace_folders)
-            ---@param folder lsp.WorkspaceFolder
-            :map(function(folder)
-              return folder.name
-            end)
-            :join('\n    ')
-        )
+        local wfolders = {} --- @type string[]
+        for _, dir in ipairs(client.workspace_folders) do
+          wfolders[#wfolders + 1] = dir.name
+        end
+        dirs_info = ('- Workspace folders:\n    %s'):format(table.concat(wfolders, '\n    '))
       else
         dirs_info = string.format(
-          '  Root directory: %s',
+          '- Root directory: %s',
           client.root_dir and vim.fn.fnamemodify(client.root_dir, ':~')
         ) or nil
       end
       report_info(table.concat({
         string.format('%s (id: %d)', client.name, client.id),
         dirs_info,
-        string.format('  Command: %s', cmd),
-        string.format('  Settings: %s', vim.inspect(client.settings, { newline = '\n  ' })),
+        string.format('- Command: %s', cmd),
+        string.format('- Settings: %s', vim.inspect(client.settings, { newline = '\n  ' })),
         string.format(
-          '  Attached buffers: %s',
+          '- Attached buffers: %s',
```

Comment:
off-topic, I guess report_info should accept a list which would be treated as list items.